test/unit/connection_string.test.ts

```diff
@@ -148,6 +148,19 @@ describe('Connection String', function () {
     expect(options.replicaSet).to.equal('123abc');
   });
 
+  context('when directionConnection is set', () => {
+    it('sets directConnection successfully when there is one host', () => {
+      const options = parseOptions('mongodb://localhost:27027/?directConnection=true');
+      expect(options.directConnection).to.be.true;
+    });
+
+    it('throws when directConnection is true and there is more than one host', () => {
+      expect(() =>
+        parseOptions('mongodb://localhost:27027,localhost:27018/?directConnection=true')
+      ).to.throw(MongoParseError, 'directConnection option requires exactly one host');
+    });
+  });
```

Comment:
Is there some spec test for this as well? I think @dariakp uncovered this at the same time as me because she was working on those

Reply:
Yes, there are two tests tagged to this ticket in uri_options.spec.test

Reply:
Yeah, you're right.  Good catch!  

If we have this covered in a spec test do you think it's worth keeping the tests I added?

Reply:
I don't think it hurts to keep our tests, but, strictly speaking, it's not really necessary since the existing spec tests do cover both of the above scenarios